Analysis/Tasks/PWGHF/HFCandidateCreator2Prong.cxx

```diff
@@ -160,6 +160,11 @@ struct HFCandidateCreator2ProngMC {
       if (RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 421, array{+kPiPlus, -kKPlus}, true, &sign) > -1) {
         result = sign * D0ToPiK;
       }
+      // Jpsi → e+e-
+      //Printf("Checking Jpsi → e+e-");
+      if (RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 443, array{+kElectron, -kElectron}, true, &sign) > -1) {
+        result = sign * JpsiToEE;
+      }
```

Comment:
- This will not work. `std::move` can be used only the last time the variable value is used.
- Wrap this block into a `if (result == N2ProngDecays)` condition. The matching should not be executed if it already matched the previous case.

See how it is done in `HFCandidateCreator3Prong.cxx`.